### PR TITLE
[RISCV] Teach fillUpExtensionSupportForSplat to handle nxvXi64 VMV_V_X_VL on RV32.

### PR DESCRIPTION
**Pull request by topperc**
A nxvXi64 VMV_V_X_VL on RV32 sign extends its 32 bit input to 64 bits. If that input is positive, the sign extend can also be considered as a zero extend.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/rvv/fixed-vectors-vror.ll [hunk 1]
```diff
@@ -1924,18 +1964,34 @@ define <8 x i64> @vror_vv_v8i64(<8 x i64> %a, <8 x i64> %b) {
...
 
 define <8 x i64> @vror_vx_v8i64(<8 x i64> %a, i64 %b) {
-; CHECK-LABEL: vror_vx_v8i64:
-; CHECK:       # %bb.0:
-; CHECK-NEXT:    vsetivli zero, 8, e64, m4, ta, ma
-; CHECK-NEXT:    vmv.v.x v12, a0
-; CHECK-NEXT:    li a0, 63
-; CHECK-NEXT:    vand.vx v16, v12, a0
-; CHECK-NEXT:    vsrl.vv v16, v8, v16
-; CHECK-NEXT:    vrsub.vi v12, v12, 0
```
Comment (lukel97) [inline llvm/test/CodeGen/RISCV/rvv/fixed-vectors-vror.ll:1934]: Looks like this causes us to miss the vrsub.vi pattern. But I think it + the subsequent vand.vx could all be done in the scalar domain?

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 2]
```diff
@@ -14477,10 +14477,22 @@ struct NodeExtensionHelper {
     // widening opcode by splatting to smaller element size.
     unsigned EltBits = VT.getScalarSizeInBits();
     unsigned ScalarBits = Op.getValueSizeInBits();
-    // Make sure we're getting all element bits from the scalar register.
-    // FIXME: Support implicit sign extension of vmv.v.x?
-    if (ScalarBits < EltBits)
+    // Make sure we're not getting all bits from the element, we need special
+    // handling.
```
Comment (lukel97) [inline llvm/lib/Target/RISCV/RISCVISelLowering.cpp:14481]: If we're not getting all bits?

**Review by lukel97 — APPROVED** [review]

File: llvm/test/CodeGen/RISCV/rvv/fixed-vectors-vror.ll (hunk 1 above)
Comment (topperc) [reply llvm/test/CodeGen/RISCV/rvv/fixed-vectors-vror.ll:1934]: I think so. I think it all happens very late and we end up going to target nodes immediately so now generic transforms fire.

**Review by preames — APPROVED** [review]
LGTM

**Pull request merged**